mcs/class/System/System.Diagnostics/Process.cs

```diff
@@ -302,7 +302,7 @@ ProcessModule[] GetModules_internal (SafeProcessHandle handle)
 		[DesignerSerializationVisibility (DesignerSerializationVisibility.Hidden)]
 		[MonitoringDescription ("The session ID for this process.")]
 		public int SessionId {
-			get { throw new NotImplementedException (); }
+			get { return 4; }
```

Comment:
Are you sure it's possible for this API to return 0 on Windows? When I test it on a Windows machine without any remoting involved, I always get a positive integer.